Tests failing with missing ceph_pool attribute in storage class OCS object

The below tests are failing because storage class OCS object does not have a ceph_pool attribute. With the change to use default storage class in tests, ceph_pool attribute is not added.

The tests has to be modified to get ceph block pool name using helpers.default_ceph_block_pool.

Affected tests:
ocs-ci/tests/manage/pv_services/test_ceph_daemon_kill_during_resource_deletion.py
ocs-ci/tests/manage/pv_services/test_resource_deletion_during_pod_pvc_deletion.py
ocs-ci/tests/manage/pv_services/test_resource_deletion_during_pvc_pod_deletion_and_io.py
ocs-ci/tests/manage/pv_services/test_daemon_kill_during_pvc_pod_deletion_and_io.py


Error Details 
AttributeError: 'OCS' object has no attribute 'ceph_pool'

```
        for pvc_name, uuid in pvc_uuid_map.items():
            if interface == constants.CEPHBLOCKPOOL:
                ret = verify_volume_deleted_in_backend(
                    interface=interface, image_uuid=uuid,
>                   pool_name=sc_obj.ceph_pool.name
                )
E               AttributeError: 'OCS' object has no attribute 'ceph_pool'
```

--- a/tests/manage/pv_services/test_ceph_daemon_kill_during_resource_deletion.py
+++ b/tests/manage/pv_services/test_ceph_daemon_kill_during_resource_deletion.py
@@ -15,8 +15,8 @@ from ocs_ci.ocs.resources.pod import (
     get_operator_pods, delete_pods
 )
 from tests.helpers import (
-    verify_volume_deleted_in_backend,
-    wait_for_resource_count_change
+    verify_volume_deleted_in_backend, wait_for_resource_count_change,
+    default_ceph_block_pool
 )
 from tests import disruption_helpers
 
@@ -161,7 +161,6 @@ class TestDaemonKillDuringPodPvcDeletion(ManageTest):
         Kill 'resource_name' daemon while deletion of PVCs/pods is progressing
         """
         pvc_objs, self.pod_objs = setup_base
-        sc_obj = pvc_objs[0].storageclass
         self.namespace = pvc_objs[0].project.namespace
         pod_functions = {
             'mds': partial(get_mds_pods), 'mon': partial(get_mon_pods),
@@ -304,11 +303,12 @@ class TestDaemonKillDuringPodPvcDeletion(ManageTest):
         log.info("Verified: PVs are deleted.")
 
         # Verify PV using ceph toolbox. Image/Subvolume should be deleted.
+        pool_name = default_ceph_block_pool()
         for pvc_name, uuid in pvc_uuid_map.items():
             if interface == constants.CEPHBLOCKPOOL:
                 ret = verify_volume_deleted_in_backend(
                     interface=interface, image_uuid=uuid,
-                    pool_name=sc_obj.ceph_pool.name
+                    pool_name=pool_name
                 )
             if interface == constants.CEPHFILESYSTEM:
                 ret = verify_volume_deleted_in_backend(

--- a/tests/manage/pv_services/test_daemon_kill_during_pvc_pod_deletion_and_io.py
+++ b/tests/manage/pv_services/test_daemon_kill_during_pvc_pod_deletion_and_io.py
@@ -15,7 +15,8 @@ from ocs_ci.ocs.resources.pod import (
 from ocs_ci.utility.utils import TimeoutSampler, ceph_health_check
 from tests.helpers import (
     verify_volume_deleted_in_backend, wait_for_resource_state,
-    wait_for_resource_count_change, verify_pv_mounted_on_node
+    wait_for_resource_count_change, verify_pv_mounted_on_node,
+    default_ceph_block_pool
 )
 from tests import disruption_helpers
 
@@ -169,7 +170,6 @@ class TestDaemonKillDuringMultipleDeleteOperations(ManageTest):
         and IO are progressing
         """
         pvc_objs, pod_objs, rwx_pod_objs = setup_base
-        sc_obj = pvc_objs[0].storageclass
         namespace = pvc_objs[0].project.namespace
 
         num_of_pods_to_delete = 10
@@ -394,11 +394,12 @@ class TestDaemonKillDuringMultipleDeleteOperations(ManageTest):
         log.info("Verified: PVs are deleted.")
 
         # Verify PV using ceph toolbox. Image/Subvolume should be deleted.
+        pool_name = default_ceph_block_pool()
         for pvc_name, uuid in pvc_uuid_map.items():
             if interface == constants.CEPHBLOCKPOOL:
                 ret = verify_volume_deleted_in_backend(
                     interface=interface, image_uuid=uuid,
-                    pool_name=sc_obj.ceph_pool.name
+                    pool_name=pool_name
                 )
             if interface == constants.CEPHFILESYSTEM:
                 ret = verify_volume_deleted_in_backend(

--- a/tests/manage/pv_services/test_resource_deletion_during_pod_pvc_deletion.py
+++ b/tests/manage/pv_services/test_resource_deletion_during_pod_pvc_deletion.py
@@ -15,8 +15,8 @@ from ocs_ci.ocs.resources.pod import (
     get_operator_pods
 )
 from tests.helpers import (
-    verify_volume_deleted_in_backend,
-    wait_for_resource_count_change
+    verify_volume_deleted_in_backend, wait_for_resource_count_change,
+    default_ceph_block_pool
 )
 from tests import disruption_helpers
 
@@ -176,11 +176,12 @@ class DisruptionBase(ManageTest):
         log.info("Verified: PVs are deleted.")
 
         # Verify PV using ceph toolbox. Image/Subvolume should be deleted.
+        pool_name = default_ceph_block_pool()
         for pvc_name, uuid in pvc_uuid_map.items():
             if interface == constants.CEPHBLOCKPOOL:
                 ret = verify_volume_deleted_in_backend(
                     interface=interface, image_uuid=uuid,
-                    pool_name=self.sc_obj.ceph_pool.name
+                    pool_name=pool_name
                 )
             if interface == constants.CEPHFILESYSTEM:
                 ret = verify_volume_deleted_in_backend(
@@ -387,7 +388,6 @@ class TestDeleteResourceDuringPodPvcDeletion(DisruptionBase):
         Delete ceph/rook pod while deletion of PVCs/pods is progressing
         """
         self.pvc_objs, self.pod_objs = setup_base
-        self.sc_obj = self.pvc_objs[0].storageclass
         self.namespace = self.pvc_objs[0].project.namespace
         self.disruptive_base(
             interface, operation_to_disrupt, resource_to_delete

--- a/tests/manage/pv_services/test_resource_deletion_during_pvc_pod_deletion_and_io.py
+++ b/tests/manage/pv_services/test_resource_deletion_during_pvc_pod_deletion_and_io.py
@@ -15,7 +15,8 @@ from ocs_ci.ocs.resources.pod import (
 from ocs_ci.utility.utils import TimeoutSampler, ceph_health_check
 from tests.helpers import (
     verify_volume_deleted_in_backend, wait_for_resource_state,
-    wait_for_resource_count_change, verify_pv_mounted_on_node
+    wait_for_resource_count_change, verify_pv_mounted_on_node,
+    default_ceph_block_pool
 )
 from tests import disruption_helpers
 
@@ -187,7 +188,6 @@ class TestResourceDeletionDuringMultipleDeleteOperations(ManageTest):
         progressing
         """
         pvc_objs, pod_objs, rwx_pod_objs = setup_base
-        sc_obj = pvc_objs[0].storageclass
         namespace = pvc_objs[0].project.namespace
 
         num_of_pods_to_delete = 10
@@ -408,11 +408,12 @@ class TestResourceDeletionDuringMultipleDeleteOperations(ManageTest):
         log.info("Verified: PVs are deleted.")
 
         # Verify PV using ceph toolbox. Image/Subvolume should be deleted.
+        pool_name = default_ceph_block_pool()
         for pvc_name, uuid in pvc_uuid_map.items():
             if interface == constants.CEPHBLOCKPOOL:
                 ret = verify_volume_deleted_in_backend(
                     interface=interface, image_uuid=uuid,
-                    pool_name=sc_obj.ceph_pool.name
+                    pool_name=pool_name
                 )
             if interface == constants.CEPHFILESYSTEM:
                 ret = verify_volume_deleted_in_backend(

